File: src/main/java/net/reflact/engine/npc/NpcData.java
package net.reflact.engine.npc;

import java.util.List;

public class NpcData {
    public String id;
    public String name;
    public String type; // PLAYER, VILLAGER, etc.
    public String skin;
    public Location location;
    public List<Interaction> interactions;

    public static class Location {
        public String world;
        public double x, y, z;
        public float yaw, pitch;
    }

    public static class Interaction {
        public String type; // DIALOGUE, QUEST_START, QUEST_COMPLETE, COMMAND
        public String value; // Text, Quest ID, Command
    }
}

File: bin/main/net/reflact/engine/npc/NpcData.kt
package net.reflact.engine.npc

class NpcData {
    var id: String = ""
    var name: String = ""
    var type: String = "" // PLAYER, ZOMBIE, etc.
    var location: Location = Location()
    var interactions: MutableList<Interaction> = ArrayList()

    class Location {
        var x: Double = 0.0
        var y: Double = 0.0
        var z: Double = 0.0
        var yaw: Float = 0.0f
        var pitch: Float = 0.0f
        var world: String = "world"
    }

    class Interaction {
        var type: String = "" // DIALOGUE, QUEST_START, SHOP
        var value: String = ""
    }
}
